Mathlib/Data/Nat/Find.lean

```diff
@@ -97,6 +97,10 @@ variable [DecidablePred q] in
 lemma find_mono (h : ∀ n, q n → p n) {hp : ∃ n, p n} {hq : ∃ n, q n} : Nat.find hp ≤ Nat.find hq :=
   Nat.find_min' _ (h _ (Nat.find_spec hq))
 
+lemma find_congr [DecidablePred q] {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
+    Nat.find hp = Nat.find hq :=
+  le_antisymm (find_mono fun _ ↦ hpq.mpr) (find_mono fun _ ↦ hpq.mp)
```

Comment:
I'm not sure this is useful; it's just
```lean
lemma find_congr [DecidablePred q] {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
    Nat.find hp = Nat.find hq := by
  congr!
  ext
  apply hpq
```
or
```lean
lemma find_congr [DecidablePred q] {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
    Nat.find hp = Nat.find hq := by
  obtain rfl : p = q := funext fun _ => propext hpq
  congr
```

That is: this doesn't use any facts about `Nat.find`, and is just a standard result about `DecidableEq`.